.github/workflows/run-checks.yml

```diff
@@ -0,0 +1,45 @@
+name: Run checks
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
+      - "[0-9]+.[0-9]+.[0-9]+rc[0-9]+"
+
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  tox:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        environment:
+          - "py36"
+          - "py37"
+          - "py38"
+          - "flake8"
+
+        include:
+          - environment: "py36"
+            python: "3.6"
+          - environment: "py37"
+            python: "3.7"
+          - environment: "py38"
+            python: "3.8"
+
+    container:
+      image: python:${{ matrix.python }}
+
```

Comment:
using `actions/setup-python` is usually faster than containerized GHA (because they're already present on disk on the runners)

Reply:
There's not a lot of activity on this repo. If you don't mind, I'd keep this as is and revise if CI times become an issue in the future.

Reply:
sure, though I expect many to see this as the "template" repository and copy what it does -- so best to use best practices

Reply:
That's fair. This CI config is not "part" of the cookiecutter template, so I don't expect people to copy this. We need to update the CI config for the cookiecutter template too.

Do you mind updating this branch with your proposed changes related to `actions/setup-python`?